homeassistant/components/sonos/speaker.py

```diff
@@ -593,8 +591,9 @@ async def async_check_activity(self, now: datetime.datetime) -> None:
 
     async def async_offline(self) -> None:
         """Handle removal of speaker when unavailable."""
-        async with self._subscription_lock:
-            await self._async_offline()
+        if self._subscription_lock:
```

Comment:
Sorry, looking at this for the first time again in a while. Is this guard necessary to make typing happy?

Reply:
Yes sir since we are deferring it from `__init__`.